### PR TITLE
3.3/develop

### PR DESCRIPTION
**Pull request by yakatz**
Update to Pull Request 29, with PSR-0 filenames

### CODE REVIEW

File: classes/Kohana/Database/MySQLi.php [hunk 1]
```diff
@@ -0,0 +1,443 @@
...
+				mysqli_errno($this->_connection));
+		}
+	}
+
+	public function query($type, $sql, $as_object = FALSE, array $params = NULL)
+	{
+		// Make sure the database is connected
+		$this->_connection or $this->connect();
+
+		if (Kohana::$profiling)
```
Comment (sergeyklay) [inline classes/Kohana/Database/MySQLi.php:173]:
``` php
if ( ! empty($this->_config['profiling']) AND Kohana::$profiling)
{
    // Benchmark this query for the current instance
    $benchmark = Profiler::start("Database ({$this->_instance})", $sql);
}
```

File: classes/Kohana/Database/MySQLi.php [hunk 2]
```diff
@@ -0,0 +1,443 @@
...
+	public function begin($mode = NULL)
+	{
+		// Make sure the database is connected
+		$this->_connection or $this->connect();
+
+		if ($mode AND ! mysqli_query("SET TRANSACTION ISOLATION LEVEL $mode", $this->_connection))
+		{
+			throw new Database_Exception(':error',
+				array(':error' => mysql_error($this->_connection)),
+				mysql_errno($this->_connection));
```
Comment (sergeyklay) [inline classes/Kohana/Database/MySQLi.php:291]:
``` php
mysqli_errno($this->_connection))
```

File: classes/Kohana/Database/MySQLi.php [hunk 3]
```diff
@@ -0,0 +1,443 @@
...
+	 */
+	public function begin($mode = NULL)
+	{
+		// Make sure the database is connected
+		$this->_connection or $this->connect();
+
+		if ($mode AND ! mysqli_query("SET TRANSACTION ISOLATION LEVEL $mode", $this->_connection))
+		{
+			throw new Database_Exception(':error',
+				array(':error' => mysql_error($this->_connection)),
```
Comment (sergeyklay) [inline classes/Kohana/Database/MySQLi.php:290]:
``` php
array(':error' => mysqli_error($this->_connection)),
```

File: classes/Kohana/Database/MySQLi.php [hunk 4]
```diff
@@ -0,0 +1,443 @@
...
+		{
+			// Set session variables
+			$variables = array();
+
+			foreach ($this->_config['connection']['variables'] as $var => $val)
+			{
+				$variables[] = 'SESSION '.$var.' = '.$this->quote($val);
+			}
+
+			mysqli_query('SET '.implode(', ', $variables), $this->_connection);
```
Comment (sergeyklay) [inline classes/Kohana/Database/MySQLi.php:93]:
``` php
mysqli_query($this->_connection, 'SET '.implode(', ', $variables));
```

File: classes/Kohana/Database/MySQLi.php [hunk 5]
```diff
@@ -0,0 +1,443 @@
...
+	 *
+	 * @param string $mode  Isolation level
+	 * @return boolean
+	 */
+	public function begin($mode = NULL)
+	{
+		// Make sure the database is connected
+		$this->_connection or $this->connect();
+
+		if ($mode AND ! mysqli_query("SET TRANSACTION ISOLATION LEVEL $mode", $this->_connection))
```
Comment (sergeyklay) [inline classes/Kohana/Database/MySQLi.php:287]:
``` php
mysqli_query($this->_connection,"SET TRANSACTION ISOLATION LEVEL $mode");
```

File: classes/Kohana/Database/MySQLi.php [hunk 6]
```diff
@@ -0,0 +1,443 @@
...
+		{
+			throw new Database_Exception(':error',
+				array(':error' => mysql_error($this->_connection)),
+				mysql_errno($this->_connection));
+		}
+
+		return (bool) mysqli_query('START TRANSACTION', $this->_connection);
```
Comment (sergeyklay) [inline classes/Kohana/Database/MySQLi.php:294]:
``` php
return (bool) mysqli_query($this->_connection, 'START TRANSACTION', $this->_connection);
```

File: classes/Kohana/Database/MySQLi.php [hunk 7]
```diff
@@ -0,0 +1,443 @@
...
+	 * Commit a SQL transaction
+	 *
+	 * @return boolean
+	 */
+	public function commit()
+	{
+		// Make sure the database is connected
+		$this->_connection or $this->connect();
+
+		return (bool) mysqli_query('COMMIT', $this->_connection);
```
Comment (sergeyklay) [inline classes/Kohana/Database/MySQLi.php:307]:
``` php
return (bool) mysqli_query($this->_connection, 'COMMIT');
```

File: classes/Kohana/Database/MySQLi.php [hunk 8]
```diff
@@ -0,0 +1,443 @@
...
+	 * Rollback a SQL transaction
+	 *
+	 * @return boolean
+	 */
+	public function rollback()
+	{
+		// Make sure the database is connected
+		$this->_connection or $this->connect();
+
+		return (bool) mysqli_query('ROLLBACK', $this->_connection);
```
Comment (sergeyklay) [inline classes/Kohana/Database/MySQLi.php:320]:
``` php
return (bool) mysqli_query($this->_connection, 'ROLLBACK');
```

**Comment by zeelot** [comment]
Added a mysqli driver to `3.4/develop` branch.

**Pull request closed without merging**